test_complete_functionality.py

```
@@ -0,0 +1,230 @@
+            return False
+
+    except Exception as e:
+        print_test(f"Async functionality: {e}", "FAIL")
+        return False
+
+
+def test_ruff_compliance():
+    """Test that code passes ruff checks."""
+    try:
+        result = subprocess.run(
+            ["ruff", "check", "mcp-server/src/", "test_server_startup.py"],
```

Ruff (0.12.2)

Starting a process with a partial executable path

(S607)